components/webext-storage/src/db.rs

```diff
@@ -105,6 +96,67 @@ impl DerefMut for StorageDb {
     }
 }
 
+/* ugh - can't call writer.close() as it consumes self
```

Comment:
I can't work out how to do this without a lot of pain. We can't use a cell, and Option seems quite painful to just handle this case. Does anyone have any ideas?

I wonder if we should consider opening an issue on rusqlite for this? I had a quick look and can't find any open or closed issues asking for this - for the sake of a single bool they might be open to a flag which just leaks the connection on failure. (Or maybe the flag arranging to call [sqlite_close_v2](https://www.sqlite.org/c3ref/close.html) would also be fine and we'd get the new behavior largely for "free" in terms of ruqslite implementation - but I think calling `sqlite_close()` probably remains best because at least you get an error telling you something went wrong but can just avoid the panic.

I made it a draft in the meantime while we discuss next steps.

Reply:
I agree we can move ahead here. It looks like the rusqlite team are going to be fine with just not doing the panic on drop, so I might wait a week for so (there's no real hurry to get this in) and if it looks like we can upgrade to a new rusqlite soon I'll just update all the comments here, and probably even the `mem::forget` hacks etc so we can better document "we try hard to close it, but things are ok if we can't"